plugins/PostProcessingPlugin/scripts/PurgeLinesAndUnload.py

```diff
@@ -0,0 +1,752 @@
+                    "enabled": true
+                },
+                "adjust_e_loc_to":
+                {
+                    "label": "    Starting E location",
+                    "description": "This is usually a negative amount and often equal to the '-Retraction Distance'.  This 'G92 E' adjustment changes where the printer 'thinks' the end of the filament is in relation to the nozzle.  It replaces the retraction that Cura adds prior to the start of 'LAYER:0'.  If retraction is not enabled then this setting has no effect.",
+                    "type": "float",
+                    "unit": "mm  ",
+                    "default_value": -6.5,
+                    "enabled": "adjust_starting_e"
+                },
+                "enable_unload":
+                {
+                    "label": "Unload filament at print end",
+                    "description": "Adds an unload script to the Ending Gcode section.  It goes in just ahead of the M104 S0.  This scripts always unloads the active extruder.  If the unload distance is greater than 150mm it will be broken into chunks to avoid tripping the excessive extrusion warning in some firmware.",
+                    "type": "bool",
+                    "default_value": false,
+                    "enabled": true
+                },
+                "unload_distance":
+                {
+                    "label": "    Unload Distance",
+                    "description": "The amount of filament to unload.  Bowden printers usually require a significant amount and direct drives not as much.",
+                    "type": "int",
+                    "default_value": 440,
+                    "unit": "mm  ",
+                    "enabled": "enable_unload"
+                },
+                "is_rectangular":
+                {
+                    "label": "Bed is rectangular",
+                    "description": "Hidden setting that disables 'purge line length' for elliptical beds.",
+                    "type": "bool",
+                    "default_value": false,
+                    "enabled": false
+                }
+            }
+        }"""
+
+    def execute(self, data):
+        # Run the selected procedures
+        # Mapping settings to corresponding methods
+        procedures = {
+            "add_purge_lines": self._add_purge_lines,
+            "move_to_start": self._move_to_start,
+            "adjust_starting_e": self._adjust_starting_e,
+            "enable_unload": self._unload_filament
+        }
+        # Run selected procedures
+        for setting, method in procedures.items():
+            if self.getSettingValueByKey(setting):
+                method(data)
+        # Format the startup and ending gcodes
+        data[1] = self._format_string(data[1])
+        data[-1] = self._format_string(data[-1])
+        return data
+
+    # Add Purge Lines to the user defined position on the build plate
+    def _add_purge_lines(self, data: str):
```

Comment:
based on the code I assume the data is a list of gcode commands, so in that case, the type should be `data: list[str]` and you would return the data as `list[str]`

In few places I saw that you just do a return without passing the data back, I'll mark them, but I haven't gone through the code completely to know if it's working or not, that comes later :P

Reply:
I think for the most part python doesn't really care about the type declaration (don't ask me how I figured out xD), but it helps during the code writing to recommend functions/operations that can be performed on it.